Kevinayres aws mp gsd (#54)
SUSE/technical-reference-documentation: kubernetes/start/aws/adoc/gs_rancher_aws-marketplace.adoc

// - what products are being highlighted
// Include an approved SUSE | Partner logo lock-up if possible
// Include any additional, relevant details
// = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =

{platform1} is a complete software stack for managing your entire Kubernetes landscape.
{productname} addresses the needs of DevOps teams, who are deploying applications with Kubernetes, and IT operations, who are responsible for delivering enterprise-critical services.
{productname} provides simple, consistent operations and unifies security, policy, and user management, whether you are running one Kubernetes cluster or thousands.

With _{offering1}_ in the {market}, you can easily install {productname}, enabling you to manage all your downstream clusters wherever a routable API address exists.

=== Scope

// = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
// Specify what this guide covers in no more than 2 sentences.
// = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =

In this guide, you learn how to install {productname} on {eks_long} using the _{offering1}_ listing in the {market}.

> From a single ephemeral appliance, you launch a three-node EKS cluster in a geography of your choosing and install {productname} in this cluster using the current rancher-stable release. 

Reviewer: ```suggestion
From a single ephemeral appliance, you launch a three-node EKS cluster in a geography of your choosing and install {productname} in this cluster using the current _rancher-stable_ release. 
```

> From a single ephemeral appliance, you launch a three-node EKS cluster in a geography of your choosing, and install {productname} in this cluster using the current rancher-stable release. 

Reviewer 2: Thank you.  I think this looks better.

Be sure to take note of the public subnet name you have chosen for your initial automation node deployment.

image::aws-mp-rancher-02.png[VPC creation, scaledwidth="85%", align="center"]
====

. Define an access policy.

.. Navigate to __Identity and Access Management__ (__IAM__) in the AWS Console at https://console.aws.amazon.com/iam/.

.. Select __Policies__ in the navigation pane on the left.

.. Click __Create policy__.

.. Select the __JSON__ tab.

.. Paste the contents of the JSON policy file at https://raw.githubusercontent.com/SUSE-Enceladus/suse-rancher-setup/main/aws-iam/iam_role.json[iam-role] into the text box.
+ 
image::aws-mp-rancher-03.png[Policy configuration, scaledwidth="85%", align="center"]

.. Click __Review policy__.

.. Enter 'RancherSetupEC2policy' for the __Name__ and, optionally, enter a __Description__.

.. Review the policy summary, then click __Create policy__.

. Create a service role.
// Kevin this step needs to be fleshed out like the one above.
.. In __IAM__, select __Roles__.

> .. Create a new Role named 'RancherSetupEC2role' with use case type 'EC2', and attach the policy 'RancherSetupEC2policy' to it.

Reviewer: ```suggestion
.. Create a new Role named 'RancherSetupEC2role' with use case type 'EC2' and attach the policy 'RancherSetupEC2policy' to it.
```

+
[TIP]
====
Be sure to use the HTTPS URL for a secure connection.
The server certificate is self-signed, so you need to accept the security exception in your browser. 

image::aws-mp-rancher-13.png[Instance public IP address, scaledwidth="75%", align="center"]
====

. Obtain the username and password for the instance.
+
[TIP]
====
The username is the EC2 Instance ID and the initial password is the AWS account ID without dashes.

You can obtain the Instance ID from the AWS console.

image::aws-mp-rancher-14.png[Instance ID, scaledwidth="75%", align="center"]
====

. Return to the {offering1} UI and click __Login__.

. Use the username and password you just obtained to log in.
+ 
image::aws-mp-rancher-15.png[Rancher Setup login, scaledwidth="75%", align="center"]

. Read the information provided, then click __Next__.
+ 
image::aws-mp-rancher-16.png[Rancher Setup information, scaledwidth="85%", align="center"]

. Wait for AWS Policy Simulator to check IAM permissions, then click __Next__.
//
+
If an error occurs at this stage, then the role was not correctly applied to the instance.
+ 
image::aws-mp-rancher-17.png[AWS Policy Simulator, scaledwidth="85%", align="center"]

> . Choose a region for deploying your {eks_long} management cluster, the click __Next__. 

Reviewer: ```suggestion
. Choose a region for deploying your {eks_long} management cluster, then click __Next__. 
```

+ 
image::aws-mp-rancher-18.png[AWS Region for EKS cluster, scaledwidth="85%", align="center"]

. Choose the size of your {productname} management cluster, then click __Next__.
+ 
image::aws-mp-rancher-19.png[Size of management cluster, scaledwidth="85%", align="center"]
+
[TIP]
====
'SMALL' works for most initial deployments.

> the instance that run the Rancher cluster can be resized when needed.

Reviewer: ```suggestion
The instance that runs the {productname} cluster can be resized when needed.
```

====

> . Enter the fully qualified domain name (FQDN) for your Rancher management server, then click __Next__.

Reviewer: ```suggestion
. Enter the fully qualified domain name (FQDN) for your {productname} management server, then click __Next__.
```

//
+
[NOTE]
====
The subdomain containing the FQDN must be preconfigured as an AWS Route53 hosted zone.
====
image::aws-mp-rancher-20.png[Rancher URL, scaledwidth="85%", align="center"]

> . Take note of the URL at which {productname} will be served, then click __Next__ to begin deployment of Rancher.

Reviewer: ```suggestion
. Take note of the URL at which {productname} will be served, then click __Next__ to begin deployment of {productname}.
```

+ 
image::aws-mp-rancher-21.png[Deploy Rancher on AWS EKS, scaledwidth="85%", align="center"]

. Deployment should take about 20 minutes.
//
+
When complete, review the provided information.
In particular, be sure to copy the initial password for the {productname} UI.
+ 
image::aws-mp-rancher-22.png[Congratulations - Rancher is ready, scaledwidth="85%", align="center"]
+
[NOTE]
====

> The __Clean resources__ function gives you a mechanism for removing your Rancher environment.

Reviewer: ```suggestion
The __Clean resources__ function gives you a mechanism for removing your {productname} environment.
```

. Click __Clean resources__.

. Click __Generate clean resources commands__.

. Scroll to the bottom of the page and download the linked zip file.

. Store this file in a safe place.
====

== Accessing {productname}

. Open a Web browser to the URL of your {productname} service.

. Enter the initial password, which you noted earlier, and click __Log in with Local User__.
+
image::aws-mp-rancher-23.png[Rancher first login, scaledwidth="85%", align="center"]

. When prompted, enter and confirm a new password for the 'admin' user, accept the terms and conditions, then click __Continue__.
+ 
image::aws-mp-rancher-24.png[Rancher new password, scaledwidth="85%", align="center"]

. You can now open your browser to the Rancher UI with the FQDN you provided during setup.
For example, https://rancher.example.com.
+
image::aws-mp-rancher-25.png[Rancher UI - Welcome, scaledwidth="85%", align="center"]

> . Terminate the temporary instance you created for the deployment automation to improve security and save cost. 

Reviewer: ```suggestion
. Terminate the temporary instance you created for the deployment automation to improve security and save costs. 
```

+ 
image::aws-mp-rancher-30.png[AWS Terminate Instance, scaledwidth="85%", align="center"]

. You can now create a 'Cloud Credential'  
+
image::aws-mp-rancher-31.png[Rancher - Create Cloud Credential, scaledwidth="75%", align="center"]

> and deploy or import downstream clusters and node-groups for workloads. 

Reviewer: ```suggestion
and deploy or import downstream clusters and node groups for workloads. 
```

====
 

== Priority support

image::rancher-prime-logo-color.svg[Rancher Prime, scaledwidth="25%", align="center"]

Elevate your Rancher landscape to https://ranchermanager.docs.rancher.com/getting-started/quick-start-guides/deploy-rancher-manager/prime[{productname2}].
{productname2} enables you with priority support, security assurances, extended lifecycles, access to focused architectures and Kubernetes advisories, and more.

Purchase {productname2} via the {market} with the _{offering2}_ listing for your geography.

image::aws-mp-rancher-34.png[Rancher Priority Support offerings, scaledwidth="85%", align="center"]

You can also purchase {productname2} via your normal AWS or SUSE sales channel or by contacting aws@suse.com directly to request a private offer.

Once activated, a billing and metering container runs in your environment that validates the support entitlement you purchased through the {market} with the quantity of managed, downstream cluster nodes.
This instantiates _{productname2}_ status for your deployment.

> If you procured {offering2} via the {market} you can get support by generating a 'supportconfig' and uploading it via the https://scc.suse.com/[SUSE Customer Center (SCC)].

Reviewer: ```suggestion
If you procured {offering2} via the {market}, you can get support by generating a 'supportconfig' and uploading it via the https://scc.suse.com/[SUSE Customer Center (SCC)].
```
